### ISSUE TITLE
Can't bootstrap gke clusters (invalid name)

### ISSUE THREAD

**Issue by afirth** [admiraltyio/multicluster-service-account#10]
```
CONSOLE_CLUSTER=gke_project-id-123456_europe-west1-d_excitingdev
WORKER_CLUSTER=gke_project-id-123456_europe-west1-d_throwawaydev
kubemcsa bootstrap --target-context $CONSOLE_CLUSTER --source-context 
```
```
created namespace "multicluster-service-account" in source cluster "gke_project-id-123456_europe-west1-d_throwawaydev"
created cluster role "service-account-import-controller-remote" in source cluster "gke_project-id-123456_europe-west1-d_throwawaydev"
kubemcsa: error: cannot bootstrap: ServiceAccount "gke_project-id-123456_europe-west1-d_excitingdev" is invalid: metadata.name: Invalid value: "gke_project-id-123456_europe-west1-d_excitingdev": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

By default, gke clusters include `_` in the name. This makes them invalid as `ObjectMeta.Name`s, so a ServiceAccount or other object can't just use the cluster name.

**Comment by adrienjt** [comment]
Thanks for filing this issue. This is new in v0.5.0, and I would have noticed if I had not also switched from GKE to k3d for my tests...

Here's my plan: if the cluster's name in the kubeconfig passes the regex check, it will be used as is. If not, `kubemcsa bootstrap` will fail and ask the user to supply another argument `--target-name`/`--source-name`, to be used as cluster name in the multicluster-service-account domain. Even if the kubeconfig cluster's name passes the regex check, those arguments will still be useful if the user wants a way to override.

**Comment by afirth** [comment]
that sounds like a much better solution than the road I was going down. I'm unclear on how this works under the hood. If `server` is the same, does the `clusterName` matter when connecting? i.e. can I work around this for now by renaming the cluster and context when I bootstrap?

**Comment by adrienjt** [comment]
You can safely rename clusters and contexts in a kubeconfig file. Make sure you refer to the new cluster name in the context. That would work around this issue. What matters when connecting is the server, certificates, etc.

**Comment by adrienjt** [comment]
The `--target-name` and `--source-name` options have been implemented in [v0.5.1](https://github.com/admiraltyio/multicluster-service-account/releases/tag/v0.5.1).

**Comment by afirth** [comment]
thanks, works great!
